packages/blockchain-wallet-v4-frontend/src/modals/Transactions/TransactionReport/index.js

```diff
@@ -30,12 +40,15 @@ class TransactionReportContainer extends React.PureComponent {
 
     return (
       <TransactionReport
+        closeAll={closeAll}
         coin={coin}
         csvData={csvData}
-        isValidStartDate={isValidStartDate}
+        filename={this.state.filename}
+        generating={this.state.generating}
         isValidEndDate={isValidEndDate}
-        onSubmit={() => this.props.actions.submitClicked(coin)}
-        closeAll={closeAll}
+        isValidStartDate={isValidStartDate}
+        onDownload={() => this.setState({ generating: false })}
+        onSubmit={() => this.generateCSV(coin)}
```

Comment:
coin could be extracted from props in generateCSV and onSubmit can be changed to `onSubmit={this.generateCSV}`